### PR TITLE
Add check_type option, to allow defaults type changes

### PR DESCRIPTION
**Pull request by krismatthews**
##### SUMMARY

I have a situation where some management software is repeatedly overwriting some defaults, but using the wrong type - e.g. it's doing something like

```bash
defaults write "com.apple.dock" "wvous-bl-corner" -string 0
```

e.g. a string "0" instead of an integer 0. This domain/key controls the actions of the bottom left hot corner (e.g. System Settings -> Display and Dock -> Hot Corner Shortcuts). When it is set as a string, it simply does not work (settings does not recognize that any value was set).

When getting said management software to stop doing this silly thing failed, I created an ansible playbook which runs  immediately after the other management software screws things up.

By default, the `osx_defaults` module will raise an error if the type of the provided `value` does not match the type of an existing defaults entry. This is a sane and useful thing to do, but in my particular use case it causes trouble. I could manually delete each default before adding it, but that feels less idempotent - and it is slightly annoying because it always triggers the handle, restarting the various apps affected.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
osx_defaults

### CODE REVIEW

File: changelogs/fragments/8173-osx_defaults-check_type.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - osx_defaults - add option ``check_types`` to enable changing the type of existing defaults on the fly.
```
Comment (felixfontein) [inline changelogs/fragments/8173-osx_defaults-check_type.yml:2]:
```suggestion
  - osx_defaults - add option ``check_types`` to enable changing the type of existing defaults on the fly (https://github.com/ansible-collections/community.general/pull/8173).
```

File: plugins/modules/osx_defaults.py [hunk 2]
```diff
@@ -50,6 +50,12 @@
     type: str
     choices: [ array, bool, boolean, date, float, int, integer, string ]
     default: string
+  check_type:
+    description:
+      - Checks if the type of the provided O(value) matches the type of an existing default.
+      - If the types do not match, raises an error.
+    type: bool
+    default: true
```
Comment (felixfontein) [inline plugins/modules/osx_defaults.py:58]:
```suggestion
    default: true
    version_added: 8.6.0
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

**Comment by felixfontein** [comment]
If nobody objects, I'll merge this in ~a week.

**Comment by felixfontein** [comment]
@krismatthews thanks for your contribution!

**Comment by krismatthews** [comment]
> @krismatthews thanks for your contribution!

@felixfontein, thank you for the review and merge. :)

**Pull request merged**